HLTriggerOffline/HeavyFlavor/src/HeavyFlavorValidation.cc

```diff
@@ -145,43 +158,29 @@ void HeavyFlavorValidation::dqmBeginRun(const edm::Run& iRun, const edm::EventSe
 		LogWarning("HLTriggerOfflineHeavyFlavor") << "Could not initialize HLTConfigProvider with process name: "<<triggerProcessName<<endl;
 		return;
 	}
-	stringstream os;
 	vector<string> triggerNames = hltConfig.triggerNames();
-	for( size_t i = 0; i < triggerNames.size(); i++) {
-		TString triggerName = triggerNames[i];
-		if (triggerName.Contains(triggerPathName)){ 
-			vector<string> moduleNames = hltConfig.moduleLabels( triggerNames[i] );
-			for( size_t j = 0; j < moduleNames.size(); j++) {
-				TString name = moduleNames[j];
-                                // MK 2016-12-09: added requirement for the module EDM type to be
-                                // EDFilter, as without it the name check can match also any EDProducer
-                                // (and would be fragile; ok I think it is still fragile, but the added
-                                // check allows PR #16792 to go forward)
-				if(name.Contains("Filter") && hltConfig.moduleEDMType(moduleNames[j]) == "EDFilter"){
-					int level = 0;
-					if(name.Contains("L1"))
-						level = 1;
-					if(name.Contains("L2"))
-						level = 2;
-					if(name.Contains("L3"))
-						level = 3;
-					if(name.Contains("mumuFilter") || name.Contains("DiMuon") || name.Contains("MuonL3Filtered") || name.Contains("TrackMassFiltered"))
-						level = 4;
-					if(name.Contains("Vertex") || name.Contains("Dz"))
-						level = 5;
-					filterNamesLevels.push_back( pair<string,int>(moduleNames[j],level) );
-					os<<" "<<moduleNames[j];
-				}
-			}
-			break;
+        for (const auto &trigName : triggerNames) {
+                // TString triggerName = trigName;
+                if (trigName.find(triggerPathName) != std::string::npos) {
+		        vector<string> moduleNames = hltConfig.moduleLabels( trigName );
+                        for (const auto &moduleName : moduleNames) {
+                                const int level = getFilterLevel(moduleName, hltConfig);
+                                if (level > 0) {
+                                        filterNamesLevels.push_back( {moduleName, level} );
+                                }
+                        }
+                        break;
 		}
 	}
-	if(filterNamesLevels.size()==0){
+
+	if(filterNamesLevels.empty()){
 		LogDebug("HLTriggerOfflineHeavyFlavor")<<"Bad Trigger Path: "<<triggerPathName<<endl;
 		return;
 	}else{
+          	stringstream os;
+                for (const auto &filters : filterNamesLevels) os << " " << filters.first;
```

Comment:
hi @tmadlener - you can avoid the string stream here - just str=str+" "+filters.first in the loop